src/modules/dataFetching/components/GenericTableWithData.tsx

```diff
@@ -150,14 +156,55 @@ const GenericTableWithData = <
   const previousQueryVariables = usePrevious(queryVariables);
   const [filterValues, setFilterValues] = useState(defaultFilterValues);
   const [sortOrder, setSortOrder] = useState<typeof defaultSortOptionProp>();
-  const [includedOptionalColumns, setIncludedOptionalColumns] = useState<
-    string[]
-  >(
-    compact(
+
+  const currentPath = useCurrentPath();
+
+  const defaultOptionalCols = useMemo(() => {
+    const storedOptionalCols = currentPath
+      ? getStoredPathParams(currentPath)?.optionalColumns
+      : undefined;
+
+    // If local storage has optional columns for this path, return them
+    if (storedOptionalCols) return storedOptionalCols;
+
+    // Otherwise, default to any optional columns that are not defaultHidden
+    return compact(
       columnsProp
         ?.filter((col) => col.optional && !col.defaultHidden)
         .map((col) => getColumnKey(col)) || []
-    )
+    );
+  }, [columnsProp, currentPath]);
+
+  // Store optional column state in BOTH search params and local storage.
+  // This is so that users can both
+  // - get the same view on their data every time they visit a table
+  // - and share/bookmark URLs that keep the same view into the data.
+  // We plan to eventually add other table controls, like sort/filter.
+  const [paramValues, setParamValues] = useSearchParamsState(
+    {
+      optionalColumns: {
+        type: 'string',
+        multiple: true,
+        default: defaultOptionalCols,
```

Comment:
This puts the column key in the URL, which can be a spaced string (from the header), or an underscored string (if column key is supplied explicitly). Is it important to clean this up?

Leads to urls that look like this:
`
/projects/:id/enrollments/households?optionalColumns=assigned_staff&optionalColumns=Last+Contact+Date
`

Reply:
Worth going through to make keys non-optional